iOS/Scenes/Library/LibraryViewController.swift

```diff
@@ -944,6 +944,25 @@ extension LibraryViewController {
         super.collectionView(collectionView, didUnhighlightItemAt: indexPath)
     }
 
+    func downloadAllAction(manga: Manga) {
+        Task {
+            let source = SourceManager.shared.source(for: manga.sourceId)
+            let chapters = (try? await source?.getChapterList(manga: manga)) ?? []
+            DownloadManager.shared.download(chapters: chapters, manga: manga)
+        }
+    }
+
+    func downloadUnreadAction(manga: Manga) {
+        Task {
+            let source = SourceManager.shared.source(for: manga.sourceId)
+            let readingHistory = await CoreDataManager.shared.getReadingHistory(sourceId: manga.sourceId, mangaId: manga.id)
+            let chapters = ((try? await source?.getChapterList(manga: manga)) ?? []).filter {
```

Comment:
probably best to use `CoreDataManager.shared.getChapters` instead https://github.com/Aidoku/Aidoku/blob/d19878092bfe2168d95e7a234b743d55d320580e/Shared/Managers/CoreData/CoreDataManager%2BChapter.swift#L51